### DISCUSSION TITLE
Allow dragging & dropping any file or folder into a sandboxed app

### DISCUSSION THREAD

**Discussion by ghost** [flatpak/xdg-desktop-portal#1249] [New Portals]
Hello, folks!

I don't know if anyone else has already had this problem, but whenever I try dragging & dropping a file into an app that accepts DnD (f.i. Amberol and Firefox), if the file or folder is not inside of the folders that the application has access, the drag & drop is simply ignored.

I use WhatsApp Web a lot, and sometimes I want to share media or documents, I can't just drag & drop. I have to go to the share menu in WA, then select the media and then confirm.

I think it would be very nice if instead of ignoring the DnD, the file was shared with the app via the documents portal. Alternatively, it could show a pop-up asking something like "Share this file with <App Name>?"

What do y'all think?

**Comment by GeorgesStavracas** [comment]
This is already the case for years, but various bugs prevented it from working properly. [These bugs have been fixed recently](https://github.com/flatpak/xdg-desktop-portal/releases/tag/1.18.2).
